include/libtorrent/bdecode.hpp

```diff
@@ -166,15 +166,18 @@ void escape_string(std::string& ret, char const* str, int len);
 struct bdecode_token
 {
 	// the node with type 'end' is a logical node, pointing to the end
-	// of the bencoded buffer.
+	// of the bencoded buffer. THe `long_string` type is for strings that are so
+	// long they need a length prefix that's longer than 8 decimal digits.
+	// these enum values need to be compatible with bdecode_node::type_t
 	enum type_t : std::uint8_t
-	{ none, dict, list, string, integer, end };
+	{ none, dict, list, string, integer, long_string, end };
 
 	enum limits_t
 	{
 		max_offset = (1 << 29) - 1,
 		max_next_item = (1 << 29) - 1,
-		max_header = (1 << 3) - 1
+		short_string_max_header = (1 << 3) - 1 + 2,
```

Comment:
```suggestion
		short_string_max_header = (1 << 3) + 1,
```

Reply:
I changed (in my mind, I simplified) the check against these upper limits, to make them also include the implied + 2

Reply:
see this check: https://github.com/arvidn/libtorrent/pull/6817/files#diff-e0c2c30bf10517b10c7e267b7717b23cbad276da4c395ab4eeaf6ba46a2f00b6R949

Reply:
-1 + 2 = + 1

Reply:
ah, this is the same reason as the other comment. the largest value stored in 3 bits is one part of the expression (`(1 << 3) - 2`), and + 2 is separate.